### PR TITLE
factories are factories

### PR DESCRIPTION
**Pull request by vigsterkr**
they are always gonna love :)

### CODE REVIEW

**Comment by vigsterkr** [comment]
this is an attempt to fix #4321

**Comment by gf712** [comment]
erf, seems like it isn't that straightforward...

**Comment by gf712** [comment]
create_object in `class_list.h` also increments the ref count

**Comment by vigsterkr** [comment]
imo that refcount++ inn create_object should be dropped... but i'm gonna test it on donbot

**Comment by vigsterkr** [comment]
this became a memory leaking cleanup branch...

File: examples/meta/src/evaluation/cross_validation_multiple_kernel_learning_weights_storage.sg [hunk 1]
```diff
@@ -21,8 +21,8 @@ combined_kernel.add("kernel_array", k_3)
 #![create_kernels]
 
 #![create_classifier]
-SVM libsvm = as_svm(machine("LibSVM"))
-Machine svm = machine("MKLClassification", kernel=combined_kernel, interleaved_optimization=False, svm=libsvm)
+SVM libsvm = machine("LibSVM")
```
Comment (vigsterkr) [inline examples/meta/src/evaluation/cross_validation_multiple_kernel_learning_weights_storage.sg]: for some reason if `as_svm` is being used here the ref counting is broken :S

File: examples/meta/src/regression/multiple_kernel_learning.sg [hunk 2]
```diff
@@ -25,8 +25,8 @@ combined_kernel.init(features_train, features_train)
 #![create_combined_train]
 
 #![train_mkl]
-SVM binary_svm_solver = as_svm(machine("SVRLight"))
-Machine mkl = machine("MKLRegression", svm=binary_svm_solver, kernel=combined_kernel, labels=labels_train)
+SVM binary_svm_solver = machine("SVRLight")
```
Comment (vigsterkr) [inline examples/meta/src/regression/multiple_kernel_learning.sg]: for some reason if `as_svm` is being used here the ref counting is broken :S

File: src/interfaces/swig/factory.i [hunk 3]
```diff
@@ -1,3 +1,28 @@
+%newobject shogun::distance(std::string name);
```
Comment (gf712) [inline src/interfaces/swig/factory.i]: doesn't it matter that it is `const std::string&`?

File: examples/meta/src/regression/multiple_kernel_learning.sg (hunk 2 above)
Comment (gf712) [reply examples/meta/src/regression/multiple_kernel_learning.sg]: broken in which way? too high or too low?

**Pull request merged**